### PR TITLE
[harmonyhub] Updates to latest harmony library, uses shared jetty client, small fi…

### LINKED ISSUE #6013: [harmonyhub] Binding loses connection after restart or bundle installs
I started using the 2.5 test bindings. Since then the HarmonyHub binding often loses connection to the hub after operations that cause reloading of bindings. All things including the bridge go offline. I currently cannot explain it in more detail. But looking at the binding source code there was only one change in the last period of time, so I guess it has to do with that change (that, btw. seems to be related).

I am using an Odroid N2. 
openjdk version "1.8.0_222"
Linux odroid 4.9.187-53 #1 SMP PREEMPT Fri Aug 9 06:12:18 -03 2019 aarch64 aarch64 aarch64 GNU/Linux

### PR DESCRIPTION
**Pull request by digitaldan**
…xes to ignore commands if offline.  This PR  takes advantage of our shared jetty instance in OH core, previously the Harmony Library was creating and managing its own instance.  This was in response to behavior noticed by a few who upgraded to our milestone release.

Fixes #6013 

Signed-off-by: digitaldan <dan@digitaldan.com>

### CODE REVIEW

File: bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyDeviceHandler.java [hunk 1]
```diff
@@ -85,14 +85,19 @@ public HarmonyDeviceHandler(Thing thing, HarmonyHubHandlerFactory factory) {
...
-            logger.warn("No such channel: {}", channelUID);
+        if (command instanceof RefreshType) {
+            // nothing to refresh
             return;
         }
 
-        if (command instanceof RefreshType) {
-            // nothing to refresh
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.warn("Hub is offline, ignoring command {} for channel {}", command, channelUID);
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyDeviceHandler.java]: reduce to debug or try to reconnect here

File: bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyHubHandler.java [hunk 2]
```diff
@@ -95,16 +95,16 @@
...
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
         logger.trace("Handling command '{}' for {}", command, channelUID);
 
-        if (!client.isConnected()) {
-            logger.warn("Cannot send command '{}' on {} because HarmonyClient is not connected", command, channelUID);
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.warn("Hub is offline, ignoring command {} for channel {}", command, channelUID);
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyHubHandler.java]: reduce to debug or try to reconnect here

File: bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyDeviceHandler.java [hunk 3]
```diff
@@ -85,14 +85,19 @@ public HarmonyDeviceHandler(Thing thing, HarmonyHubHandlerFactory factory) {
...
+            return;
+        }
+
+        Channel channel = getThing().getChannel(channelUID.getId());
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyDeviceHandler.java]: Why did you add this here? You should assume that the framework makes sure that commands are only sent for existing channels.

Comment (digitaldan) [reply bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyDeviceHandler.java]: My guess is that i was using that channel variable at some point during testing, and needed to check it's null-ness to appease the nullable annotations, but since i'm not using it for anything, its very much not necessary.

**Comment by digitaldan** [comment]
PR updated
